lib/discovery/updater.go

```diff
@@ -156,35 +160,46 @@ func (u *updater) onModifiedOrAddedEvent(serviceObj service, service serviceKey,
 			portMatcher: port.Port,
 		}
 
-		// TODO(bplotka): Add support for TLS between (even insecure for now).
-		if port.Name == "http" || strings.HasPrefix(port.Name, "http-") {
+		scheme := getScheme(port.Name)
+		switch scheme {
+		case httpScheme, httptlsScheme:
 			if hostMatcherOverride != "" {
 				foundRoute.nameMatcher = hostMatcherOverride
 			}
 
 			// We need to avoid specific ports if possible, since Golang removes the port from request.URL.Port() for default ports,
 			// even when user specifies http://<host>:80 or https://<host>:443 explicitly. As a result we convert default
 			// ports to NO port, which mean host-wide routing.
-			if port.Port == 80 {
+			if scheme == httpScheme && port.Port == 80 {
+				foundRoute.portMatcher = 0
+			}
+			if scheme == httptlsScheme && port.Port == 443 {
```

Comment:
I would like to move these checks to some func e.g.
```
if isDefaultPort(scheme, port.Port) {
    foundRoute.portMatcher = 0
}
```
but it's up to you

Reply:
that makes sense